shell/pages/auth/login.vue

```diff
@@ -179,19 +118,76 @@ export default {
   },
 
   async fetch() {
+    const { firstLoginSetting } = await this.loadInitialSettings();
     const { value } = await this.$store.dispatch('management/find', { type: MANAGEMENT.SETTING, id: SETTING.BANNERS });
+    const drivers = await this.$store.dispatch('auth/getAuthProviders');
+    const providers = sortBy(drivers.map((x) => x.id), ['id']);
+    const hasLocal = providers.includes('local');
+    const hasOthers = hasLocal && !!providers.find((x) => x !== 'local');
 
-    this.customLoginError = JSON.parse(value).loginError;
-  },
+    if ( hasLocal ) {
+      // Local is special and handled here so that it can be toggled
+      removeObject(providers, 'local');
+    }
```

Comment:
I love how we modify const lists. I really think immutable should take place of referentially const practically everywhere.

Reply:
Our utilities are aimed at doing that, so it may be worth starting the conversation from that point since it's simpler.
For the record, I think me you and @rak-phillip are all in for that and we can even use ESLint for it, including Vue specific cases.